### PR TITLE
edit food route

### PR DESCRIPTION
**Pull request by zeinjendeya**
- created a query that takes the food category, user id, the food amount, and food id.
- created the controller for the route
- added the route in the route file.

#40

### CODE REVIEW

**Review by khamis-hesham — APPROVED** [review]

File: server/controllers/editFood.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+const editFood = async (req, res, next) => {
+  const {
+    user: { id },
+  } = req;
+  const { categoryId, foodId } = req.params;
+  const { quantityInGrams } = req.body;
+  try {
+    if (!id) {
+      throw Boom.unauthorized('you are not logged!');
+    }
```
Comment (tomduggan85) [inline server/controllers/editFood.js:13]: I think you could check for negative quantityInGrams values here (or non-numeric ones too) and throw a 400 if there are any problems found.

Comment (zeinjendeya) [reply server/controllers/editFood.js:13]: like this?
```
if (quantityInGrams < 0 || typeof quantityInGrams !== 'number') {
      throw Boom.badRequest('invalid amount of grams');
    }
```

Comment (tomduggan85) [reply server/controllers/editFood.js:13]: Yep! If you want, you could move the typeof check first, and the greater than zero second - I’m not sure what javascript will do if you try to check whether a non-number is greater than 0 😅

Comment (lina-jamal) [reply server/controllers/editFood.js:13]: you can just use validate schema from joi that we use
to check validation to quantityInGrams values

Comment (lina-jamal) [reply server/controllers/editFood.js:13]: you can just use validate schema from joi that we use to check validation to quantityInGrams values

**Review by hassanelnajjar — APPROVED** [review]

**Review by khamis-hesham — APPROVED** [review]

**Review by lina-jamal — APPROVED** [review]

**Review by imansedky5 — DISMISSED** [review]

File: server/database/queries/editFoodGrams.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const connection = require('../config/connection');
+
+const editFoodGrams = (userID, categoryID, foodID, quantityGrams) => {
+  const sql = {
+    text:
+      'UPDATE UserFoodRelation SET amount_in_grams=$1 WHERE users_id=$2 AND food_id=$3 AND food_category_id=$4 RETURNING amount_in_grams',
```
Comment (hassanelnajjar) [inline server/database/queries/editFoodGrams.js:6]: You need to check the date of the update operation.
```sql
UPDATE UserFoodRelation SET amount_in_grams=$1 WHERE users_id=$2 AND food_id=$3 AND food_category_id=$4 and created_at=current_date RETURNING amount_in_grams

```

**Review by hassanelnajjar — DISMISSED** [review]

**Review by hassanelnajjar — APPROVED** [review]

**Review by lina-jamal — APPROVED** [review]

**Pull request merged**